wallet/src/de/schildbach/wallet/ui/WalletFragment.kt

```diff
@@ -0,0 +1,342 @@
+        }
+        join_dashpay_action_space.visibility = join_dashpay_action.visibility
+    }
+
+    private fun showHideSecureAction() {
+        secure_action.visibility = if (config.remindBackupSeed) View.VISIBLE else View.GONE
+        secure_action_space.visibility = secure_action.visibility
+    }
+
+    private fun handleVerifySeed() {
+        val checkPinSharedModel = ViewModelProviders.of(this)[CheckPinSharedModel::class.java]
+        checkPinSharedModel.onCorrectPinCallback.observe(viewLifecycleOwner, Observer<Pair<Int?, String?>?> { data ->
+            if (data?.second != null) {
+                startVerifySeedActivity(data.second!!)
+            }
+        })
+        show(requireActivity(), 0)
+    }
+
+    private fun handleScan(clickView: View?) {
+        ScanActivity.startForResult(requireActivity(), clickView, MainActivity.REQUEST_CODE_SCAN)
+    }
+
+    private fun startVerifySeedActivity(pin: String) {
+        val intent: Intent = createIntent(requireContext(), pin)
+        startActivity(intent)
+    }
+
+    private fun handlePaste() {
+        var input: String? = null
+        if (clipboardManager!!.hasPrimaryClip()) {
+            val clip = clipboardManager!!.primaryClip ?: return
+            val clipDescription = clip.description
+            if (clipDescription.hasMimeType(ClipDescription.MIMETYPE_TEXT_URILIST)) {
+                val clipUri = clip.getItemAt(0).uri
+                if (clipUri != null) {
+                    input = clipUri.toString()
+                }
+            } else if (clipDescription.hasMimeType(ClipDescription.MIMETYPE_TEXT_PLAIN)
+                    || clipDescription.hasMimeType(ClipDescription.MIMETYPE_TEXT_HTML)) {
+                val clipText = clip.getItemAt(0).text
+                if (clipText != null) {
+                    input = clipText.toString()
+                }
+            }
+        }
+        if (input != null) {
+            handleString(input, R.string.scan_to_pay_error_dialog_title, R.string.scan_to_pay_error_dialog_message)
+        } else {
+            InputParser.dialog(requireContext(), null, R.string.scan_to_pay_error_dialog_title, R.string.scan_to_pay_error_dialog_message_no_data)
+        }
+    }
+
+    private fun handleString(input: String, errorDialogTitleResId: Int, cannotClassifyCustomMessageResId: Int) {
+        object : StringInputParser(input, true) {
+            override fun handlePaymentIntent(paymentIntent: PaymentIntent) {
+                SendCoinsInternalActivity.start(requireContext(), paymentIntent, true)
+            }
+
+            override fun handlePrivateKey(key: PrefixedChecksummedBytes) {
+                SweepWalletActivity.start(requireContext(), key, true)
+            }
+
+            @Throws(VerificationException::class)
+            override fun handleDirectTransaction(tx: Transaction) {
+                //TODO: Uncomment
+                //application.processDirectTransaction(tx);
```

Comment:
I suppose this goes here: `walletApplication.processDirectTransaction(tx)`

Reply:
Fixed on https://github.com/dashevo/dash-wallet/pull/462/commits/e794ccefe9c91675c7cb0896a503b17325019754